src/Disks/DiskEncrypted.cpp

```diff
@@ -41,34 +54,122 @@ namespace
         }
     }
 
+#if USE_AWS_S3
+
+    String getDecryptedKeyUsingAwsKms(const Poco::Util::AbstractConfiguration & config, const String & config_prefix, const String & key_path)
+    {
+        auto logger = getLogger("DiskEncrypted");
+
+        String key_id_arn_path = key_path + "[@key_arn]";
+        if (!config.has(key_id_arn_path))
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Missing key_arn for key_aws {}", key_path);
+        String key_id_arn = config.getString(key_id_arn_path);
+        String role_arn;
+        String role_arn_path = key_path + "[@role_arn]";
+        if (config.has(role_arn_path))
+            role_arn = config.getString(role_arn_path);
+        String encrypted_key = config.getString(key_path);
+
+        auto client_config = S3::ClientFactory::instance().createClientConfiguration(
+            Aws::Environment::GetEnv("AWS_DEFAULT_REGION"),
```

Comment:
why we're not taking them from the configuration?

Reply:
Because we're not configuring the AWS secrets in the config file, but through env variables. We could introduce a new field within the disk as I did for `aws_kms_endpoint` and `no_sign_request` that adds that. But it'd be redundant and the `S3CredentialsProviderChain` [will use env vars anyway](https://github.com/ClickHouse/ClickHouse/blob/7721d4fb37e215be2ddbd2366e64312708387061/src/IO/S3/Credentials.cpp#L692-L715) to authenticate with the credentials provider we need.